tfe/resource_tfe_policy_set.go

```diff
@@ -43,10 +45,50 @@ func resourceTFEPolicySet() *schema.Resource {
 				ConflictsWith: []string{"workspace_external_ids"},
 			},
 
+			"vcs_repo": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"policy_ids"},
+				MinItems:      1,
+				MaxItems:      1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"identifier": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"branch": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"ingress_submodules": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+
+						"oauth_token_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
```

Comment:
Something about the ordering 🤐

Reply:
That's fine. It should be `TypeList` anyway unless we need to avoid duplicates. I have been avoiding sets in that case as well too to avoid the side-effects of hashing and the kind of un-intuitive nature of the duplicate squishing with sets (in addition to how fragile it is really, given that the hashing implies an exact match).

There is a new block type coming to TF 0.12 that will allow you to specify blocks by key, which is what I will probably be favoring for cases where duplicates need to be avoided.